sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/BuiltinApproxCountDistinctSqlAggregator.java

```diff
@@ -177,4 +177,10 @@ private static class BuiltinApproxCountDistinctSqlAggFunction extends SqlAggFunc
       );
     }
   }
+
+  private boolean validateInputComplexTypeName(ColumnType columnType)
+  {
+    return Objects.equals(columnType.getComplexTypeName(), HyperUniquesAggregatorFactory.TYPE.getComplexTypeName()) ||
+           Objects.equals(columnType.getComplexTypeName(), HyperUniquesAggregatorFactory.PRECOMPUTED_TYPE.getComplexTypeName());
```

Comment:
This doesn't allow `UNKNOWN_COMPLEX` types. Should it support them as well? It looks inconsistent given that other aggregators allow `UNKNOWN_COMPLEX`

Reply:
I'm not sure on this. Will check with Clint / Zoltan on all the pending review comments where I'm lacking context to make a decision.

Reply:
I'm a big fan of only allowing what's absolutely necessary.